packaging/debian/storagenode.config

```diff
@@ -144,41 +146,65 @@ case "$STATE" in
             else
                   go_back
             fi
-      ;;
+	    ;;
       "$IP_ADDRESS")
-            db_input high storagenode/external_addr || true
-            if db_go; then
-                  db_get storagenode/external_addr
-                  ip_port="$RET"
-                  ip="${ip_port%:*}"
-                  port="${ip_port##*:}"
-
-                  if ! valid_ip $ip ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  elif ! valid_port $port ;  then
-                        error_ask_continue storagenode/external_addr/not_ip_port_continue
-                  fi
-                  next_state
-            else
-                  go_back
-            fi
-      esac
+          db_input high storagenode/external_addr || true
+          if db_go; then
+              db_get storagenode/external_addr
+              ip_port="$RET"
+              ip="${ip_port%:*}"
+              port="${ip_port##*:}"
+
+              if ! valid_ip $ip ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              elif ! valid_port $port ;  then
+                  error_ask_continue storagenode/external_addr/not_ip_port_continue
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$STORAGEDIR")
+          db_input high storagenode/storagedir || true
+          if db_go; then
+              db_get storagenode/storagedir
+              storagedir="$RET"
+
+	      if [ ! -d "$storagedir" ]; then
+		  error_ask_continue storagenode/storagedir/does_not_exist
+              fi
+              next_state
+          else
+              go_back
+          fi
+	  ;;
+      "$DISKSPACE")
+          db_input high storagenode/diskspace || true
+          if db_go; then
+              db_get storagenode/diskspace
+              diskspace="$RET"
+	      if [[ ! "$diskspace" =~ ^[0-9]+[GTgt]{1} ]]; then
+		  error_ask_continue storagenode/diskspace/wrong_format
+	      else
+		  space_amount="${diskspace%*[GTgt]}"
+		  if [ "$space_amount" -lt "500" ]; then
+		      error_ask_continue storagenode/diskspace/below_minimum
+		  fi
+		  if [ "$ERR_STATE" != "$ERR_BACK" ]; then
+		      space_unit="${diskspace##*[0-9]}"
+		      freespace=`df --output=avail -B "$space_unit" "$storagedir" |tail -n 1`
+		      freespace_amount="${freespace%*[GT]}"
+
+		      if [ "$freespace_amount" < "$space_amount" ]; then
+			  error_ask_continue storagenode/diskspace/below_specified
```

Comment:
This won't catch all likely situations: for example if a user asks for `1T` of space to be used, but they only have 50GB available, both `$freespace_amount` and `$space_amount` will be `1` (apparently df rounds up). Maybe it would be better to calculate the actual number of bytes desired and bytes free in all cases?